src/Workspaces/Core/Portable/Workspace/Solution/StateChecksums.cs

```diff
@@ -124,17 +107,19 @@ public static SolutionStateChecksums Deserialize(ObjectReader reader)
         if (searchingChecksumsLeft.Count == 0)
             return;
 
-        if (hintProject != null)
+        if (assetHint.ProjectId != null)
         {
-            var projectState = state.GetProjectState(hintProject);
+            var projectState = state.GetProjectState(assetHint.ProjectId);
             if (projectState != null &&
                 projectState.TryGetStateChecksums(out var projectStateChecksums))
             {
-                await projectStateChecksums.FindAsync(projectState, searchingChecksumsLeft, result, cancellationToken).ConfigureAwait(false);
+                await projectStateChecksums.FindAsync(projectState, assetHint.DocumentId, searchingChecksumsLeft, result, cancellationToken).ConfigureAwait(false);
             }
         }
         else
         {
+            Contract.ThrowIfTrue(assetHint.DocumentId != null);
```

Comment:
this must be true.  because if we had a DocumentId hint, then we would have had it's corresponding ProjectId hint.